### PR TITLE
Run diagnostics for all prognostic runs

### PR DESCRIPTION
**Pull request by nbren12**
Trigger the diagnostics for all prognostic runs. Previously this was only hooked into the end-to-end, train-diag-progs workflow.

### CODE REVIEW

**Comment by nbren12** [comment]
I manually triggered the E2E tests in this argo job: integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107

**Comment by nbren12** [comment]
I also rebased this on master.

**Comment by nbren12** [comment]
The online diagnostics failed: 
```
✖ integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107  main
 ├-✖ nudge-to-fine-run                                             prognostic-run/prognostic-run                                                                                               child 'integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107-4065173114' failed
 | ├---✔ resolve-output-url                                        resolve-output-url/resolve-output-url  integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107-3099984103  4s
 | ├---✔ prepare-config                                            prepare-config                         integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107-660667410   12s
 | ├---✔ run-model                                                 run-fv3gfs/run-fv3gfs
 | |   ├---✔ create-run                                            create-run                             integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107-1894391460  30s
 | |   └---✔ run-first-segment                                     run-all-segments
 | |       ├---✔ append-segment                                    append-segment                         integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107-3070488323  3m
 | |       ├---✔ increment-segment                                 increment-count                        integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107-2922169645  4s
 | |       └---✔ run-next-segment                                  run-all-segments
 | |           ├---✔ append-segment                                append-segment                         integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107-3564084628  5m
 | |           ├---✔ increment-segment                             increment-count                        integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107-3169853512  3s
 | |           └---○ run-next-segment                              run-all-segments                                                                                                            when '2 < 2' evaluated false
 | └---✖ online-diags                                              prognostic-run-diags/diagnostics                                                                                            child 'integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107-421948929' failed
 |     └---✖ compute-reduced-diagnostics                           compute-reduced-diagnostics            integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-3107-421948929   4m        failed with exit code 1
 ├-✔ resolve-output-url
```

**Comment by nbren12** [comment]
I'm having second thoughts about this. The E2E tests rely on the prognostic run exit code to detect failures, but this will mask that effect. I suppose we could handle this the same way as the upload for failed runs works.

**Comment by oliverwm1** [comment]
> I'm having second thoughts about this. The E2E tests rely on the prognostic run exit code to detect failures, but this will mask that effect. I suppose we could handle this the same way as the upload for failed runs works.

Could just remove the continue-on-fail and have the diagnostics only get automatically computed for runs that finish successfully. Otherwise, yeah we could do some stuff in the e2e test argo workflow to make sure it detects prognostic run failures.

**Comment by nbren12** [comment]
> > The online diagnostics failed for the nudged run
> 
> what was the error message?

Argo deleted the logs before I had a chance to look. I suspect your right.

**Comment by nbren12** [comment]
I manually triggered these integration tests: integration-test-7ff7223b2a0720aac7ff7c5eecadab3da2eae7f3-2ed1

**Comment by nbren12** [comment]
That set of tests passed successfully, and produced all of the expected data.

File: workflows/argo/README.md [hunk 1]
```diff
@@ -71,6 +71,8 @@ sequential segments.
 | `segment-count`      | (optional) Number of prognostic run segments; default "1"                     |
 | `cpu`                | (optional) Number of cpus to request; default "6"                             |
 | `memory`             | (optional) Amount of memory to request; default 6Gi                           |
+| `online-diags-flags` | (optional) `flags` for `prognostic-run-diags` workflow                        |
+| `online-diags`       | (optional) Run online diagostics if "true; default "true"                     |
```
Comment (oliverwm1) [inline workflows/argo/README.md:75]:
```suggestion
| `online-diags`       | (optional) Run online diagostics if "true"; default "true"                     | 
```

**Review by oliverwm1 — APPROVED** [review]
I'm approving, although would be curious to know if you actually tried a case where the prognostic run failed.

File: workflows/argo/README.md (hunk 1 above)
Comment (nbren12) [reply workflows/argo/README.md:75]: I didn't try introducing a fault in the code. I think it will work...

**Pull request merged**